src/components/clusterexplorer/node.ts

```diff
@@ -81,7 +85,29 @@ export type ClusterExplorerNode =
     ClusterExplorerHelmHistoryNode |
     ClusterExplorerCustomNode;
 
+export type ClusterExplorerNodeV2 =
+    ClusterExplorerMessageNode |
+    ClusterExplorerClusterNode |
+    ClusterExplorerContextNode |
+    ClusterExplorerResourceFolderNode |
+    ClusterExplorerGroupingFolderNode |
+    ClusterExplorerResourceNode |
+    ClusterExplorerConfigurationValueNode |
+    ClusterExplorerHelmReleaseNode |
+    ClusterExplorerHelmHistoryNode |
+    ClusterExplorerCustomNode;
+
 export class ClusterExplorerNodeImpl {
     readonly nodeCategory = KUBERNETES_EXPLORER_NODE_CATEGORY;
+    customizedTreeItem?: vscode.TreeItem;
     constructor(readonly nodeType: KubernetesExplorerNodeType) {}
+    getTreeItem(): vscode.TreeItem | Thenable<vscode.TreeItem> {
+        if (this.customizedTreeItem) {
```

Comment:
It took me a while to remember why i have done this. And i'm not even sure i recall everything. BTW, by reading this comment https://github.com/Azure/vscode-kubernetes-tools/pull/767#issuecomment-691699061 it looks like there was an issue when the tree was drawn. The method `getTreeItem` inside the explorer.ts returns the node customized if there are customisers. The problem is, for example, that if i have two nodes at the same level [A, B], if there are no customisers they are drawn in the correct order [A,B] but if the first node has one or more customisers and the second not, eventually, the second node will appear in the first position because the other node has been customized in the meantime and the tree re-drawn it [B, A].

So what i have done is ... the  `getTreeItem` inside the explorer.ts returns the node as it is, then if there are customisers they are called async and when they finish the refresh method make a new call to the `getTreeItem`. Inside `node.ts` i then check if we have a customization to prevent vscode from reloading all children and re-drawing all tree and changing the order of our nodes.

Any other idea to do this?

I think that this could be the reason why you see a loading spinner when the tree loads. Maybe you have a customisers that took a while to add its customization. ??